### PR TITLE
Support call-by-name for Box's openOrThrowException.

### LINKED ISSUE #1781: Call Box.openOrThrowException parameter by name
**Suggestion to the `Box` API:** make  `openOrThrowException` parameter called by name instead of by value. This will allow to avoid calculation of the “justification” expression for full box.

https://github.com/lift/framework/blob/master/core/common/src/main/scala/net/liftweb/common/Box.scala#L421

### PR DESCRIPTION
**Pull request by farmdawgnation**
Closes #1781.

cc @serioga

### CODE REVIEW

**Comment by Shadowfiend** [comment]
We should be able to spec-test the call-by-name nature of the parameter by passing something that computes the value + has a side effect, and confirming that the side effect only happens if the `Box` is non-`EmptyBox`.

Lookin' good though!

**Comment by farmdawgnation** [comment]
Added examples as requested.

File: core/common/src/test/scala/net/liftweb/common/BoxSpec.scala [hunk 1]
```diff
@@ -289,6 +305,22 @@ class BoxSpec extends Specification with ScalaCheck with BoxGenerator {
     "define a 'asA' method returning Empty" in {
       Empty.asA[Double] must_== Empty
     }
+
+    "invoke a call-by-name parameter to openOrThrowException" in {
+      var sideEffect = false
+      def sideEffecting = {
+        sideEffect = true
+        "This shouldn't have been invoked."
```
Comment (Shadowfiend) [inline core/common/src/test/scala/net/liftweb/common/BoxSpec.scala]: May be worth changing this to "this *should* have been invoked" hehe.

Comment (farmdawgnation) [inline core/common/src/test/scala/net/liftweb/common/BoxSpec.scala]: Sure.

**Comment by farmdawgnation** [comment]
Heh codacy is complaining about duplication in my specs. :unamused:

**Comment by Shadowfiend** [comment]
I'm 👍 for this. Once Lift 3 goes final ofc ;)

**Comment by Shadowfiend** [comment]
Still 👍 here so let's roll 'er in.

**Pull request merged**